openverse_catalog/dags/providers/provider_api_scripts/stocksnap.py

```diff
@@ -28,175 +28,165 @@
+            self._page_counter += 1
+            return True
+        else:
+            return False
+
+    def get_batch_data(self, response_json):
+        """
+        Take an API response and return the list of records.
+        """
+        # Based on the sample test file, using json.load on the file will return a list.
+        if isinstance(response_json, list) and len(response_json) > 0:
+            return response_json
         return None
-    creator, creator_url = _get_creator_data(media_data)
-    metadata = _get_metadata(media_data)
-    tags = _get_tags(media_data)
-
-    return {
-        "title": title,
-        "creator": creator,
-        "creator_url": creator_url,
-        "foreign_identifier": foreign_id,
-        "foreign_landing_url": foreign_landing_url,
-        "image_url": image_url,
-        "filesize": _get_filesize(image_url),
-        "filetype": "jpg",
-        "height": height,
-        "width": width,
-        "license_info": license_info,
-        "meta_data": metadata,
-        "raw_tags": tags,
-        "category": prov.DEFAULT_IMAGE_CATEGORY[PROVIDER],
-    }
+
+    def get_record_data(self, data):
+        """
+        Parse out the necessary information (license info, urls, etc) from the record
+        data (single image dict-like thing) into a dictionary.
+        """
+        try:
+            foreign_id = data["img_id"]
+        except (TypeError, KeyError, AttributeError):
+            return None
+
+        foreign_landing_url = f"https://{HOST}/photo/{foreign_id}"
+
+        image_url, width, height = self._get_image_info(data)
+        if image_url is None:
+            logger.info("Found no image url.")
+            logger.info(f"{json.dumps(data, indent=2)}")
+            return None
+
+        title = self._get_title(data)
+        if title is None:
+            logger.info("Found no image title.")
+            logger.info(f"{json.dumps(data, indent=2)}")
+            return None
+
+        creator, creator_url = self._get_creator_data(data)
+        metadata = self._get_metadata(data)
+        tags = self._get_tags(data)
+        filesize = self._get_filesize(image_url)
+        lic = get_license_info("https://creativecommons.org/publicdomain/zero/1.0/")
+
+        return {
+            "title": title,
+            "creator": creator,
+            "creator_url": creator_url,
+            "foreign_identifier": foreign_id,
+            "foreign_landing_url": foreign_landing_url,
+            "image_url": image_url,
+            "filesize": filesize,
+            "filetype": "jpg",
+            "height": height,
+            "width": width,
+            "license_info": lic,
+            "meta_data": metadata,
+            "raw_tags": tags,
+            "category": prov.DEFAULT_IMAGE_CATEGORY[PROVIDER],
+        }
+
+    @staticmethod
+    def _get_image_info(item):
+        width = ClevelandDataIngester._get_int_value(item, "img_width")
```

Comment:
`_get_int_value` is useful, but I'd prefer to pull it out of `ClevelandDataIngester` and into a shared `util` module if it's going to be reused elsewhere. Another alternative would be to add this cleaning step to the `MediaStore` (maybe [here](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/storage/media.py#L98)?) so that all providers get it automatically.

Reply:
Ah, you're right, since those fields are media type specific it shouldn't go in `MediaStore`! But maybe we could extend `clean_media_metadata` in the `ImageStore`?

If we go that route, we'd have the advantage of simplifying even more provider script code and automatically getting the validation for all image providers (even ones that haven't been refactored yet). But for the same reason it's probably best to pull that into a separate PR so we don't increase the scope of this one. If you think that's a good idea, I think we can pull the logic out of this script for now (since the code currently in production does not do this validation) and wait to add it for all providers.

Reply:
Yeah, that makes total sense to me. I'll do that.

Reply:
By "that" I mean just taking out the validation for now, though extending `clean_media_metadata` in the `ImageStore` makes total sense, and I wish I had thought of it. :)